application/adamantine.cc

```diff
@@ -197,6 +200,35 @@ int main(int argc, char *argv[])
       return 0;
     }
 
+    // Abusing the database to pull out the output dir when writing the output
+    if (map.count("output-dir") == 1)
+    {
+      try
+      {
+        std::string outdir =
+            std::filesystem::absolute(map["output-dir"].as<std::string>())
+                .string() +
+            std::filesystem::path::preferred_separator;
+        std::filesystem::create_directories(outdir);
+
+        database.put("post_processor.output_dir", outdir);
+      }
+      catch (std::runtime_error const &exception)
+      {
+        std::cerr << exception.what() << std::endl;
+        return 0;
+      }
+    }
+    else if (database.count("post_processor.output_dir") == 0)
+    {
+      database.put("post_processor.output_dir", "");
+    }
+
+    // Make adamantine behave a bit better and not lock up if not in cwd of
+    // current input file
+    std::filesystem::current_path(
+        std::filesystem::path(filename).parent_path());
```

Comment:
@stvdwtt can you check that the workflow code and the INTERSECT adapters still work with this change.

Reply:
@stvdwtt any update on this?

Reply:
@cadkin, @Rombur -- sorry for the delay. I'll do some testing with the control driver by the end of the week to make sure it doesn't cause issues (or issues that are easily addressed).

Reply:
@stvdwtt anything new on this?

Reply:
Sorry @cadkin and @Rombur for taking so long on this. I finally tested it with the driver. I was able to modify the driver to work with this change pretty straightforwardly.

Unless I'm missing something, it appears that we can no longer give the input file as just the name of the file, it has to be a path. Before the standard approach was to have the input file in the CWD and point to the input file with `-i input.info`. Now, I believe it has to be `-i ./input.info`. 

@cadkin, do you know if there's any simple way to support the older syntax without the `./`? If not, we'd have to update the documentation as part of this PR.

Reply:
> do you know if there's any simple way to support the older syntax without the ./?

If it's not possible, we could add a new option maybe something like `-p` (`-input_path`) with the new behavior.

Reply:
@Rombur, that's a good idea. I'm checking right now to see if it's possible to autodetect a relative path and treat it differently, but if that doesn't work I like adding a new option

Reply:
```suggestion
        if (std::filesystem::path(filename).has_parent_path()){
          std::filesystem::path(filename).parent_path());
        }
```

This seems to have the intended effect of only changing the CWD if `filename` has a parent path.

@cadkin, would this work for you?

Reply:
Interesting, I would have expected that `std::filesystem::path` would have treated a path the same regardless of how it was input... very strange.

It seems that using `std::filesystem::absolute` instead forces the `parent_path()` to be defined since it can't be missing in an absolute path. Give my latest commit a try and let me know!

Reply:
This works! Approving and merging now.